### PR TITLE
Fixed the incorrect configure option and now we can compile binaries for...

### PR DESCRIPTION
**Pull request by shinnya**
Hi, all.
I found that `configure_x86_64.sh` caused a configure error and fixed it.
Please check my changes on your machine.
# Environment

```
shinya_yamaoka@syamaoka% uname -a
Darwin syamaoka.local 13.4.0 Darwin Kernel Version 13.4.0: Sun Aug 17 19:50:11 PDT 2014; root:xnu-2422.115.4~1/RELEASE_X86_64 x86_64 i386 MacBookPro8,1 Darwin

shinya_yamaoka@syamaoka% clang -v
Apple LLVM version 6.0 (clang-600.0.54) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin13.4.0
Thread model: posix

shinya_yamaoka@syamaoka% xcodebuild -version
Xcode 6.1
Build version 6A1052d
```
# Problem
## configure_x86_64.sh generates a configure error

Here is the configure error which I got:

```
shinya_yamaoka@syamaoka% ./configure_x86_64.sh 
checking for ICU version numbers... release 54.1, library 54.1, unicode version 7.0
checking build system type... x86_64-apple-darwin13.4.0
checking host system type... i686-apple-darwin11
checking whether to build debug libraries... no
checking whether to build release libraries... yes
checking for i686-apple-darwin11-clang... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
checking whether the C compiler works... no
configure: error: in `/Users/shinya_yamaoka/repos/mine/icu-ios-shinnya/build-x86_64':
configure: error: C compiler cannot create executables
See `config.log' for more details
```

Open `build-x86_64/config.log` and the error messages like below are shown in the log:

```
configure:2962: checking whether the C compiler works
configure:2984: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator8.1.sdk -I/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator8.1.sdk/usr/include/ -I./include/ -arch x86_64 -miphoneos-version-min=5.0 -I/Users/shinya_yamaoka/repos/mine/icu-ios-shinnya/icu/source/common/ -I/Users/shinya_yamaoka/repos/mine/icu-ios-shinnya/icu/source/tools/tzcode/   -stdlib=libc++ -L/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator8.1.sdk/usr/lib/ -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator8.1.sdk -Wl,-dead_strip -miphoneos-version-min=5.0 -lstdc++ conftest.c  >&5
ld: entry point (start) undefined.  Usually in crt1.o for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

On Mac OSX, the ld entry point is not `_start` but `_main`, so ld command shows an error message. But this is not the root cause of the configure error.
It is the issue that we give a incorrect configure option in `configure_x86_64.sh`.
# Solution

According to http://source.icu-project.org/repos/icu/icu/trunk/readme.html#HowToCrossCompileICU, we should not specify `--host` configure option on generating `x86_64` static libraries. `--host` option must be specified only on generating binaries for iOS. Then, when we give `--host` configure option, we must give `--with-cross-build` configure option too.

Best Regards.

### CODE REVIEW

File: configure_x86_64.sh [hunk 1]
```diff
@@ -7,12 +6,12 @@ ICU_FLAGS="-I$ICU_PATH/source/common/ -I$ICU_PATH/source/tools/tzcode/ "
 
 export CXX="$DEVELOPER/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++"
 export CC="$DEVELOPER/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang"
-export CFLAGS="-isysroot $SDKROOT -I$SDKROOT/usr/include/ -I./include/ -arch $ARCH -miphoneos-version-min=5.0 $ICU_FLAGS"
-export CXXFLAGS="-stdlib=libc++ -std=c++11 -isysroot $SDKROOT -I$SDKROOT/usr/include/ -I./include/ -arch $ARCH -miphoneos-version-min=5.0 $ICU_FLAGS"
-export LDFLAGS="-stdlib=libc++ -L$SDKROOT/usr/lib/ -isysroot $SDKROOT -Wl,-dead_strip -miphoneos-version-min=5.0 -lstdc++"
```
Comment (shinnya) [inline configure_x86_64.sh:12]: `configure` failed when I defined environment variabels related to iOS. It seems that giving such configure options are regarded as cross compiling. In the first place, we only have to create static libraries for x86_64 here. Right?

File: configure_i386.sh [hunk 2]
```diff
@@ -7,12 +6,12 @@ ICU_FLAGS="-I$ICU_PATH/source/common/ -I$ICU_PATH/source/tools/tzcode/ "
 
 export CXX="$DEVELOPER/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++"
 export CC="$DEVELOPER/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang"
-export CFLAGS="-isysroot $SDKROOT -I$SDKROOT/usr/include/ -I./include/ -arch $ARCH -miphoneos-version-min=7.0 $ICU_FLAGS"
-export CXXFLAGS="-stdlib=libc++ -std=c++11 -isysroot $SDKROOT -I$SDKROOT/usr/include/ -I./include/ -arch $ARCH -miphoneos-version-min=7.0 $ICU_FLAGS"
-export LDFLAGS="-stdlib=libc++ -L$SDKROOT/usr/lib/ -isysroot $SDKROOT -Wl,-dead_strip -miphoneos-version-min=7.0 -lstdc++"
```
Comment (shinnya) [inline configure_i386.sh:12]: See the comment posted to `configure_x86_64.sh` to understand why these codes will be removed.

**Comment by zhm** [comment]
Removing the `isysroot` makes it use your system header files, e.g. `/usr/include` instead of the iOS SDK's version of the system header files in `/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator8.1.sdk/usr/include/`. So, it would be building a static library against the OSX header files instead of for the iOS simulator header files.

I think the problem is the `-miphoneos-version-min=5.0` setting. Changing `-miphoneos-version-min=5.0` to `-miphoneos-version-min=7.0` fixes the problem. `_start` is correct (runs before `_main`), but I don't think it's linking the crt because the SDK version issue so it can't find `_start`.

**Comment by zhm** [comment]
Can you try changing the original version to `-miphoneos-version-min=7.0` (to match the other arch configure scripts) on your machine and see if it fixes the problem?

**Comment by shinnya** [comment]
@zsm
Thank you for your kindness.  I'll try `-miphoneos-version-min=7.0`.

**Comment by shinnya** [comment]
I tried `-miphoneos-version-min=7.0` and all my builds finished successfully. Your expectation is true and my solution is incorrect at all. Thank you.

**Comment by shinnya** [comment]
@zsm
Shall I recreate another pull request(changing `-miphoneos-version-min=5.0` to `-miphoneos-version-min=7.0` in `configure_x86_64.sh`) or just close this pull request?

**Comment by zhm** [comment]
Great! Can you close this pull request and create a new pull request with just that change? Thanks!

**Pull request closed without merging**